alphabase/io/tempmmap.py

```diff
@@ -65,12 +132,24 @@ def array(shape: tuple, dtype: np.dtype) -> np.ndarray:
         A tuple with the shape of the array.
     dtype : type
         The np.dtype of the array.
+    tmp_dir_abs_path : str, optional
+        If specified the memory mapped array will be created in this directory.
+        An absolute path is expected.
+        Defaults to None. If not specified the global TEMP_DIR_NAME location will be used.
 
     Returns
     -------
     type
         A writable temporary mmapped array.
     """
+
+    global TEMP_DIR_NAME
+
+    # redefine the temporary directory if a new location is given otherwise read from global variable
+    # this allows you to ensure that the correct temp directory location is used when working with multiple threads
+    if tmp_dir_abs_path is not None:
+        TEMP_DIR_NAME = _change_temp_dir_location(tmp_dir_abs_path)
```

Comment:
do we need this assignment here? after all, `TEMP_DIR_NAME` is set in `_change_temp_dir_location()` .. either remove the assignment here (and move the `global` to `_change_temp_dir_location` ?